### PR TITLE
Implementing headers support (not overriding old headers)

### PR DESCRIPTION
**Pull request by mario1105**

### CODE REVIEW

File: src/httpClient.js [hunk 1]
```diff
@@ -1,98 +1,97 @@
...
+    },
+
+    runRequest: function (url, opts) {
+        return this.request(url, opts).run()
+    },
+
+    run: function () {
+        return this.value('onBeforeRun')(this).doRun()
+    },
+    doRun: function () {
```
Comment (mario1105) [inline src/httpClient.js]: changes here

File: src/httpClient.test.js [hunk 2]
```diff
@@ -2,304 +2,319 @@ const originalHttpClient = require(`./httpClient`)
...
+            done()
+        }.bind(this))
     })
 
-    this.httpClient.onBeforeRun(beforeRunCallback).runRequest(`/assert_options`)
-  })
-
-  it(`allows POST`, (done) => {
-    this.fetch.post(`/post`, "Hello")
+    it(`allows passing a header`, (done) => {
```
Comment (mario1105) [inline src/httpClient.test.js]: new test

File: src/httpClient.js (hunk 1 above)
Comment (manuelmorales) [reply src/httpClient.js]: Mario, please don't change the indentation of the whole file. It's very difficult to know what changed. And if we finally agree that four spaces are better, at least commit that change separately so changes can be compared individually.

**Review by manuelmorales — CHANGES_REQUESTED** [review]
Sorry Mario, but I need the changes of indentation either gone or in a separate commit so I can review this properly Mario.

**Review by manuelmorales — APPROVED** [review]

**Pull request merged**